runtime/onert/frontend/base_loader/include/base_loader.h

```diff
@@ -175,30 +182,32 @@ template <typename LoaderDomain, typename SpecificLoader> class BaseLoader
 template <typename LoaderDomain, typename SpecificLoader>
 void BaseLoader<LoaderDomain, SpecificLoader>::BaseLoader::loadFromFile(const char *file_path)
 {
-  std::ifstream stream(file_path, std::fstream::in | std::fstream::binary);
-
-  if (!stream)
+  int fd = open(file_path, O_RDONLY);
+  if (fd < 0)
   {
-    std::string msg = "Failed to open file `";
-    msg += file_path;
-    msg += "`";
-    throw std::runtime_error{msg};
+    throw std::runtime_error("Failed to open file " + std::string(file_path));
   }
 
-  stream.seekg(0, stream.end);
-  auto size = stream.tellg();
-  stream.seekg(0, stream.beg);
-
-  _buffer.resize(size);
-  stream.read(_buffer.data(), size);
+  struct stat file_stat;
+  if (fstat(fd, &file_stat) != 0)
+  {
+    throw std::runtime_error("Fstat failed or file " + std::string(file_path) +
+                             " is not a regular file");
+  }
+  int size = file_stat.st_size;
 
-  stream.close();
+  // Map model file into memory region
+  _base = static_cast<char *>(mmap(NULL, size, PROT_READ, MAP_PRIVATE, fd, 0));
```

Comment:
Why didn't you use _base as local variable?

Reply:
@ragmani Following PR will use that value in `loadOperand` funtion to partially unmaps memory region for weight values [ed2e01f](https://github.com/Samsung/ONE/pull/2097/commits/ed2e01f35382bed2e5a8e3abeab10de5eff0bf37)

Reply:
I see. Thanks.